### PR TITLE
[MIG] account_invoice_tax_required: Migration to 13.0

### PR DESCRIPTION
**Pull request by albariera**
This PR finishes the migration from #756 

Tests have been improved

### CODE REVIEW

**Review by etobella — APPROVED** [review]
LGTM :+1: Functional and code Review

**Comment by etobella** [comment]
@kittiu @captivea-dcr @victoralmau @jbaudoux

**Review by HaraldPanten — APPROVED** [review]

File: account_invoice_tax_required/models/account_move.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                _("%s\n%s") % (_("No Taxes Defined!"), "\n".join(x for x in errors))
+            )
+
+    def action_post(self):
+        # Always test if it is required by context
+        force_test = self.env.context.get("test_tax_required")
+        skip_test = any(
+            (
+                # It usually fails when installing other addons with demo data
+                self.with_user(SUPERUSER_ID)
```
Comment (victoralmau) [inline account_invoice_tax_required/models/account_move.py]:
```suggestion
                self.sudo()
```

File: account_invoice_tax_required/models/account_move.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# Copyright 2015 - Camptocamp SA - Author Vincent Renaville
+# Copyright 2016 - Tecnativa - Angel Moya <odoo@tecnativa.com>
+# Copyright 2019 - Tecnativa - Pedro M. Baeza
+# Copyright 2019 - Punt Sistemes - Juan Vicente Pascual
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import SUPERUSER_ID, _, models
```
Comment (victoralmau) [inline account_invoice_tax_required/models/account_move.py]:
```suggestion
from odoo import _, models
```

File: account_invoice_tax_required/models/account_move.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                        ("demo", "=", True),
+                    ]
+                ),
+                # Avoid breaking unaware addons' tests by default
+                config["test_enable"],
+            )
+        )
+        if self.type != "entry" and (force_test or not skip_test):
+            self._test_invoice_line_tax()
+        return super(AccountMove, self).action_post()
```
Comment (victoralmau) [inline account_invoice_tax_required/models/account_move.py]:
```suggestion
        return super().action_post()
```

**Review by victoralmau — CHANGES_REQUESTED** [review]

**Comment by albariera** [comment]
Thank you for the review. Comments attended :+1:

**Review by AaronHForgeFlow — APPROVED** [review]
Code review LGTM

**Comment by AaronHForgeFlow** [comment]
@victoralmau If you consider your comments are attended can you provide some feedback?

**Comment by victoralmau** [comment]
> Thank you for the review. Comments attended +1

Sorry, the changes in my comments are not applied, right?

**Comment by albariera** [comment]
Sorry, I did it wrong. Now  the comments are applied! @victoralmau

**Review by victoralmau — APPROVED** [review]

**Pull request merged**